crates/biome_graphql_parser/src/parser/definitions/operation.rs

```diff
@@ -0,0 +1,137 @@
+use crate::parser::{
+    directive::DirectiveList,
+    parse_error::{expected_any_selection, expected_name, expected_selection_set},
+    parse_name, GraphqlParser,
+};
+use biome_graphql_syntax::{
+    GraphqlSyntaxKind::{self, *},
+    T,
+};
+use biome_parser::{
+    parse_lists::ParseNodeList, parse_recovery::ParseRecovery, parsed_syntax::ParsedSyntax,
+    prelude::ParsedSyntax::*, token_set, Parser, TokenSet,
+};
+
+pub(crate) const OPERATION_TYPE: TokenSet<GraphqlSyntaxKind> =
+    token_set![T![query], T![mutation], T![subscription]];
+
+/// https://spec.graphql.org/October2021/#sec-Language.Operations.Query-shorthand
+#[inline]
+pub(crate) fn parse_operation_definition(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_operation(p) {
+        return Absent;
+    }
+
+    if p.at_ts(OPERATION_TYPE) {
+        // TODO: parse variables
+        let m = p.start();
+        {
+            let m = p.start();
+            p.bump_ts(OPERATION_TYPE);
+            m.complete(p, GRAPHQL_OPERATION_TYPE);
+        }
+
+        // we don't need diagnostic here, because name is optional
+        parse_name(p).ok();
+
+        DirectiveList.parse_list(p);
+        parse_selection_set(p).or_add_diagnostic(p, expected_selection_set);
+
+        Present(m.complete(p, GRAPHQL_OPERATION_DEFINITION))
+    } else {
+        parse_selection_set(p)
+    }
+}
+
+#[inline]
+pub(crate) fn is_at_operation(p: &mut GraphqlParser<'_>) -> bool {
+    p.at_ts(OPERATION_TYPE) || is_at_selection_set(p)
+}
+
+#[inline]
+fn parse_selection_set(p: &mut GraphqlParser) -> ParsedSyntax {
+    let m = p.start();
+    p.expect(T!['{']);
+    SelectionList::new().parse_list(p);
+    p.bump(T!['}']);
```

Comment:
Yeah, it looks good!

Sorry, I forgot to mention that we also need to use `expect` here. It seems we need to check the `at/at_ts` token and call bump to ensure that we don't encounter a panic in production.

```suggestion
    p.expect(T!['}']);
```

Reply:
I understand. I didn't notice bump panic if the token was not found